### PR TITLE
NestedFolders: Enable new nested folder picker by default for nested folders

### LINKED ISSUE #65745: Build it

### PR DESCRIPTION
**Pull request by joshhunt**
Introduces the new folder picker UI that uses the full nested folder hierarchy, enabled by default when the `nestedFolders` feature flag is also enabled.

At the moment the new picker is used in new Browse Dashboards UI, Save Dashboard, and Dashboard Settings. Future pull requests will enable it more broadly.

https://github.com/grafana/grafana/assets/46142/c5df2b41-fe44-4d14-895b-406d2b75a7a6

Fixes https://github.com/grafana/grafana/issues/65745

### CODE REVIEW

**Comment by joshhunt** [comment]
Note - we cannot merge this until we've moved to the new wrapper FolderPicker component in the save dashboard form and dashboard settings settings.

**Comment by joshhunt** [comment]
e2e tests needs updating. 

probably need to add e2e selectors to the new folder picker also

**Review by ashharrison90 — APPROVED** [review]
lgtm 👍

File: pkg/services/featuremgmt/toggles_gen_test.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ func TestFeatureToggleFiles(t *testing.T) {
 	t.Run("check registry constraints", func(t *testing.T) {
 		for _, flag := range standardFeatureFlags {
 			if flag.Expression == "true" && flag.Stage != FeatureStageGeneralAvailability {
-				t.Errorf("only stable features can be enabled by default.  See: %s", flag.Name)
+				t.Errorf("only FeatureStageGeneralAvailability features can be enabled by default.  See: %s", flag.Name)
```
[the remark this answers is not in the document]
Comment (joshhunt) [reply pkg/services/featuremgmt/toggles_gen_test.go:34]: I changed this manually...

[the remark this answers is not in the document]
Comment (ashharrison90) [reply pkg/services/featuremgmt/toggles_gen_test.go:34]: ah whoops, i misread. thought this was a generated file 🤦

**Pull request merged**